beeai/jira_issue_fetcher/tests/unit/test_jira_issue_fetcher.py

```diff
@@ -0,0 +1,393 @@
+import asyncio
+import json
+import os
+import sys
+import time
+
+import pytest
+import requests
+from flexmock import flexmock
+
+from jira_issue_fetcher import JiraIssueFetcher
+from common.models import Task, TriageInputSchema, RebaseInputSchema, BackportInputSchema, RebaseOutputSchema, BackportOutputSchema, ClarificationNeededData, NoActionData, ErrorData, RebaseData, BackportData
+from common.constants import JiraLabels, RedisQueues
+from common.utils import redis_client
+
+
+@pytest.fixture
+def mock_env_vars(monkeypatch):
+    """Mock environment variables."""
+    monkeypatch.setenv('JIRA_URL', 'https://jira.test.com')
+    monkeypatch.setenv('JIRA_TOKEN', 'test_token')
+    monkeypatch.setenv('REDIS_URL', 'redis://localhost:6379')
+    monkeypatch.setenv('QUERY', 'filter = "Jotnar_1000_packages"')
+
+@pytest.fixture
+def fetcher(mock_env_vars):
+    """Create a JiraIssueFetcher instance with mocked environment."""
+    return JiraIssueFetcher()
+
+class MockAsyncContextManager:
+    """A proper async context manager for testing."""
+
+    def __init__(self, mock_redis):
+        self.mock_redis = mock_redis
+
+    async def __aenter__(self):
+        return self.mock_redis
+
+    async def __aexit__(self, exc_type, exc_val, exc_tb):
+        pass
+
+@pytest.fixture
+def mock_redis_context():
+    """Create a mock Redis context manager for testing."""
+    import sys
+    from jira_issue_fetcher import redis_client
+
+    # Create mock Redis object
+    mock_redis = flexmock()
+
+    # Create a proper async context manager
+    mock_context_manager = MockAsyncContextManager(mock_redis)
+
+    # Mock the redis_client function in the jira_issue_fetcher module where it's imported
+    flexmock(sys.modules['jira_issue_fetcher']).should_receive('redis_client').and_return(mock_context_manager)
+
+    return mock_redis, mock_context_manager
```

Comment:
This seems overly compilcated, wouldn't something like this work?

```suggestion
@pytest.fixture
def mock_redis_context():
    """Create a mock Redis context manager for testing."""
    mock_redis = flexmock()
    
    @asynccontextmanager
    async def mock_context_manager(*_, **__):
        yield mock_redis
    
    flexmock(jira_issue_fetcher).should_receive("redis_client").and_return(mock_context_manager)

    return mock_redis, mock_context_manager
```

Reply:
I tried your suggestion and it doesn't seem to work.
I agree this is complex, it took me a while to create this mock objects. But I think they are all needed. I need to mock the result of the `__aenter__`  method (for some of the tests but not all). This should be why we need a more complex mock object and the `asynccontextmanager` you suggested is not working.

Reply:
`__aenter__` returns the context variable, which is what an `@asynccontextmanager`-decorated generator does when yielding. I'm not saying the suggested code will work as it is, it was just an example, but I don't see why it would be impossible to do it this way.

Reply:
I can try to do some experiments and suggest a working code (unless I fail :slightly_smiling_face:).